src/mlang/mpp_ir/mpp_ir_to_bir.ml

```diff
@@ -10,11 +10,27 @@ end)
 
 type translation_ctx = {
   new_variables : Mir.Variable.t StringMap.t;
-  variables_used_as_inputs : unit Mir.VariableMap.t;
+  variables_used_as_inputs : Mir.VariableDict.t;
+  rule_instances : Bir.rule Bir.RuleMap.t;
 }
 
 let emtpy_translation_ctx : translation_ctx =
-  { new_variables = StringMap.empty; variables_used_as_inputs = Mir.VariableMap.empty }
+  {
+    new_variables = StringMap.empty;
+    variables_used_as_inputs = Mir.VariableDict.empty;
+    rule_instances = Bir.RuleMap.empty;
+  }
+
+let ctx_join ctx1 ctx2 =
+  {
+    new_variables =
+      (* CR keryan : I assume there shouldn't be conflict there, kept the behavior of
```

Comment:
Indeed, there shouldn't be any conflict. What happens if you replace `Some v2` by `assert false` and run the tests again ?

Reply:
By conflict I meant cases where `v1` and `v2` would be different, but using this function is unavoidable.
`ctx_join` is used to merge contexts of the branches of conditional expression, each branch is evaluated with the same entry context, including `new_variables` whose intersection will most likely not be empty.

Reply:
So the test would be `assert (Mir.Variable.compare v1 v2 = 0)` (assuming comparing the ids is enough), and there is indeed no issues on the tests.